lib/child_process.js

```diff
@@ -161,15 +168,40 @@ function exec(command, options, callback) {
 }
 
 const customPromiseExecFunction = (orig) => {
-  return (...args) => {
+  return (file, options = {}) => {
     let resolve;
     let reject;
     const promise = new Promise((res, rej) => {
       resolve = res;
       reject = rej;
     });
 
-    promise.child = orig(...args, (err, stdout, stderr) => {
+    if (typeof options !== 'object') {
+      return Promise.reject(
+        new ERR_INVALID_ARG_TYPE(
+          'options',
+          'Object',
+          options));
+    }
+
+    const { signal } = options;
+    const hasSignal = signal != null;
+    const isNotValidSignal = typeof signal !== 'object' ||
+      !('aborted' in signal);
+
+    if (hasSignal) {
+      if (isNotValidSignal) return Promise.reject(
+        new ERR_INVALID_ARG_TYPE(
+          'options.signal',
+          'AbortSignal',
+          signal));
+
```

Comment:
There needs to be a check to see if the `AbortSignal` has already been tripped before spawning the process and creating the Promise. If it has been tripped, then a rejected Promise should be returned immediately.